### ISSUE TITLE
loss=nan ,use my own data

### ISSUE THREAD

**Issue by louielu1027** [rksltnl/Deep-Metric-Learning-CVPR16#6]
i use my own face datas，and train on my pre-train model,but the loss=nan. i use my own net with your loss layer. i set the bithsize 64 when i  use your code to generate the face LMDB file. i don't know why. 
thank you very much!

**Comment by SummerTrains** [comment]
Hi! I have met the same problem, loss=NAN !
But It is not difficult to solve it!
Just Debug!
Good Luck!

**Comment by louielu1027** [comment]
@SummerTrains i know，i have debug  for a long time,but it doesn't work.   so can you explain more detail?

**Comment by SummerTrains** [comment]
in the code,lifted_structed_softmax_loss.cpp, 0/0 may exist !

however, my loss is still very high, ~11.0 , I think it is still possible to decrease the loss, and I'm still trying !

Good luck for you and me !

At 2016-08-15 11:09:35, "louielu1027" notifications@github.com wrote:

@SummerTrains i know，i have debug for a long time,but it doesn't work. so can you explain more detail?

**Comment by BeSlower** [comment]
Same error in my experiments. I use my own data, inception-v3 network and a pre-trained model trained on ImageNet.
I already tried to decrease the learning rate to a very small value or added a normalization layer before the loss layer, but neither of them works.
I have no idea what's going on.

**Comment by louielu1027** [comment]
@BeSlower my QQ:693491925 ,we can discuss this problem together

**Comment by SummerTrains** [comment]
@BeSlower  in the loss layer implementation,  some numeric problems, for example, 0/0, sqrt(negative value) may exist !
Even though, my loss is still very high!
 keep  connection and  discussion!

**Comment by daf11865** [comment]
@louielu1027  @BeSlower  @SummerTrains 
Hi, have you guys successfully solved the nan problem?
I want to do metric learning with category-level. Hence I use PASCAL detection dataset(20 classes) and combine the ROIpooling layer. I'm now struggling in using ZF pretrained model.

I add norm2 and modify several code in Lift...cpp to prevent nan,
but like @SummerTrains  said, even not encountering nan problem, the loss won't decrease.
btw, in iteration 17xx, the nan problem still appear...

**Comment by AIHGF** [comment]
@daf11865 Hi, have you successfully solved the nan problem? how do you modify the lifted loss.cpp? I also found sqrt(negative value) existed as @SummerTrains said.
